kvbc/tools/db_editor/include/kv_blockchain_db_editor.hpp

```diff
@@ -0,0 +1,486 @@
+  }
+};
+
+struct GetLastStateTransferBlockID {
+  bool read_only = true;
+  std::string description() const {
+    return "getLastStateTransferBlockID\n"
+           " Returns the last state transfer block ID. If there is no state transfer returns n/a";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &) const {
+    auto blockId = adapter.getLastStatetransferBlockId();
+    return toJson("lastStateTransferBlockID", blockId ? std::to_string(*blockId) : "n/a"s);
+  }
+};
+
+struct GetLastBlockID {
+  bool read_only = true;
+  std::string description() const {
+    return "getLastBlockID\n"
+           " Returns the last block ID. Either reachable or state transfer";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &) const {
+    return toJson("lastBlockID", getLatestBlockId(adapter));
+  }
+};
+
+struct GetRawBlock {
+  bool read_only = true;
+  std::string description() const {
+    return "getRawBlock BLOCK-ID\n"
+           "  Returns a serialized raw block (encoded in hex).";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &args) const {
+    if (args.values.empty()) {
+      throw std::invalid_argument{"Missing BLOCK-ID argument"};
+    }
+    const auto raw_block = adapter.getRawBlock(toBlockId(args.values.front()));
+    return toJson("rawBlock",
+                  raw_block ? concordUtils::vectorToHex(categorization::RawBlock::serialize(*raw_block)) : "n/a"s);
+  }
+};
+
+struct GetRawBlockRange {
+  bool read_only = true;
+  std::string description() const {
+    return "getRawBlockRange BLOCK-ID-START BLOCK-ID-END\n"
+           "  Returns a list of serialized raw blocks (encoded in hex) in the [BLOCK-ID-START, BLOCK-ID-END) range.";
+  }
+
+  std::string execute(const KeyValueBlockchain &adapter, const CommandArguments &args) const {
+    if (args.values.size() < 2) {
+      throw std::invalid_argument{"Missing or invalid block range"};
+    }
+    const auto end = toBlockId(args.values[1]);
+    if (end == 0) {
+      throw std::invalid_argument{"Invalid BLOCK-ID-END value"};
+    }
+    const auto first = toBlockId(args.values[0]);
+    const auto last = std::min(end - 1, getLatestBlockId(adapter));
+    if (first > last) {
+      throw std::invalid_argument{"Invalid block range"};
+    }
+    auto raw_blocks = std::vector<std::pair<std::string, std::string>>{};
+    for (auto i = first; i <= last; ++i) {
+      const auto raw_block = adapter.getRawBlock(i);
+      if (!raw_block) {
+        throw NotFoundException{"Couldn't find a block by ID = " + std::to_string(i)};
+      }
+      raw_blocks.emplace_back("rawBlock" + std::to_string(i),
+                              concordUtils::vectorToHex(categorization::RawBlock::serialize(*raw_block)));
+    }
+    return toJson(raw_blocks);
+  }
+};
+
+struct GetBlockInfo {
+  bool read_only = true;
```

Comment:
I realise this is not part of this PR, but we can use it to change all `read_only` members in all commands to being `const`.